### PR TITLE
KM: Moving addresses from message to PTT request containing this message

### PR DESCRIPTION
**Pull request by moriaab**
This PR removes the addresses from a message signed by the Worker add adds them to the request.
The tests are adjusted accordingly.

### CODE REVIEW

File: enigma-principal/app/src/esgx/keys_keeper_u.rs [hunk 1]
```diff
@@ -33,11 +33,10 @@ extern "C" {
 /// let response = get_enc_state_keys(enclave.geteid(), request, nonce, None).unwrap();
 /// ```
 #[logfn(DEBUG)]
-pub fn get_enc_state_keys(eid: sgx_enclave_id_t, request: StateKeyRequest, epoch_nonce: U256, sc_addrs: Option<&[ContractAddress]>) -> Result<StateKeyResponse, Error> {
+pub fn get_enc_state_keys(eid: sgx_enclave_id_t, request: StateKeyRequest, epoch_nonce: U256, sc_addrs: &[ContractAddress]) -> Result<StateKeyResponse, Error> {
```
Comment (AvishaiW) [inline enigma-principal/app/src/esgx/keys_keeper_u.rs:36]: why did you remove the option from the addresses? from my understanding, there is still the case in which no addresses are sent to the km (at the beginning of an epoch), isn't it?

File: enigma-principal/app/src/boot_network/keys_provider_http.rs [hunk 2]
```diff
@@ -115,22 +116,19 @@ impl PrincipalHttpServer {
             Some(block_number) => epoch_provider.find_epoch(block_number.try_into()?)?,
             None => epoch_provider.find_last_epoch()?,
         };
+        let addresses = &request.addresses;
         let msg = PrincipalMessage::from_message(&request.get_data()?)?;
```
Comment (AvishaiW) [inline enigma-principal/app/src/boot_network/keys_provider_http.rs]: this seems like it's only used in the else statement, it seems smarter to put it there.

File: enigma-principal/app/src/boot_network/keys_provider_http.rs [hunk 3]
```diff
@@ -115,22 +116,19 @@ impl PrincipalHttpServer {
...
-                let epoch_addrs = Self::find_epoch_contract_addresses(&request, &msg, &epoch_state)?;
-                get_enc_state_keys(*epoch_provider.eid, request, epoch_state.nonce, Some(&epoch_addrs))?
+                Self::find_epoch_contract_addresses(&request, &msg, &epoch_state)?
             }
-            _ => return Err(RequestValueErr {
-                request: METHOD_GET_STATE_KEYS.to_string(),
-                message: "Request data not found in PrincipalMessage".to_string(),
-            }.into()),
         };
+        let response = get_enc_state_keys(*epoch_provider.eid, request, epoch_state.nonce, &addrs)?;
```
Comment (AvishaiW) [inline enigma-principal/app/src/boot_network/keys_provider_http.rs:132]: is this suppose to get request or msg?

**Review by AvishaiW — APPROVED** [review]
LGTM!

File: enigma-principal/app/src/esgx/keys_keeper_u.rs (hunk 1 above)
Comment (moriaab) [reply enigma-principal/app/src/esgx/keys_keeper_u.rs:36]: It is not. This function always gets addresses: or addresses from input, or all epoch addresses.

File: enigma-principal/app/src/boot_network/keys_provider_http.rs (hunk 3 above)
Comment (moriaab) [reply enigma-principal/app/src/boot_network/keys_provider_http.rs:132]: Request as it uses also the signature

File: enigma-principal/app/src/boot_network/keys_provider_http.rs (hunk 2 above)
Comment (moriaab) [reply enigma-principal/app/src/boot_network/keys_provider_http.rs]: Thanks, will fix.

File: enigma-principal/app/src/esgx/keys_keeper_u.rs (hunk 1 above)
Comment (moriaab) [reply enigma-principal/app/src/esgx/keys_keeper_u.rs:36]: I will fix the test that gets empty vector of addresses. Thank you @AvishaiW, it is a very good catch!
